selfservice/hook/web_hook.go

```diff
@@ -0,0 +1,187 @@
+package hook
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/google/go-jsonnet"
+	"github.com/ory/kratos/selfservice/flow/registration"
+	"github.com/ory/kratos/selfservice/flow/verification"
+	"github.com/ory/kratos/session"
+	"github.com/ory/kratos/x"
+	"io"
+	"net/http"
+)
+
+var _ registration.PostHookPostPersistExecutor = new(WebHook)
+var _ verification.PostHookExecutor = new(WebHook)
+
+type (
+	webHookDependencies interface {
+		x.LoggingProvider
+	}
+
+	emptyAuthConfig struct {}
+
+	basicAuthConfig struct {
+		User     string
+		Password string
+	}
+
+	apiKeyConfig struct {
+		Name  string
+		Value string
+		In    string
+	}
+
+	AuthConfig interface {
+		apply(req *http.Request)
+	}
+
+	Auth struct {
+		Type       string
+		AuthConfig AuthConfig      `json:"-"`
+		RawConfig  json.RawMessage `json:"config"`
+	}
+
+	webHookConfig struct {
+		Method string
+		Url    string
+		Body   string
+		Auth   Auth
+	}
+
+	WebHook struct {
+		r webHookDependencies
+		c json.RawMessage
+	}
+)
+
+func (c *emptyAuthConfig) apply(_ *http.Request) {}
+
+func (c *webHookConfig) Unmarshal(bytes []byte) error {
+	if err := json.Unmarshal(bytes, c); err != nil {
```

Comment:
This is the place I meant in the PR comment related to how configuration parsing happens and which should be changed.